api/src/core/usecases/importFromSource.ts

```diff
@@ -27,7 +27,20 @@ export const importFromHALSource: (dbApi: DbApiV2) => (agentEmail: string) => Pr
             const index = dbSoftwaresNames.indexOf(software.title_s[0]);
 
             if (index != -1) {
-                return dbSoftwares[index].softwareId;
+                console.log(dbSoftwares[index].externalId, " == ", software.docid);
+                if (dbSoftwares[index].externalId === software.docid) {
+                    return dbSoftwares[index].softwareId;
+                } else {
+                    // New HAL notice version, need to update
+                    let newSoft = await halRawSoftwareToSoftwareForm(software);
+                    await dbApi.software.update({
+                        softwareSillId: dbSoftwares[index].softwareId,
+                        formData: newSoft,
+                        agentId: agentId
+                    });
+
+                    return dbSoftwares[index].softwareId;
+                }
```

Comment:
on peut alléger un peu  (sans else car on return) et tu as un `let` au lieu de `const`

```suggestion
                if (dbSoftwares[index].externalId === software.docid) {
                    return dbSoftwares[index].softwareId;
                } 
                    // New HAL notice version, need to update | On garde ce commentaire ??
                  const newSoft = await halRawSoftwareToSoftwareForm(software);
                  await dbApi.software.update({
                        softwareSillId: dbSoftwares[index].softwareId,
                        formData: newSoft,
                        agentId: agentId
                    });

                  return dbSoftwares[index].softwareId;
                }
```